test/units/testsuite-83.sh

```diff
@@ -3,6 +3,13 @@
 set -eux
 set -o pipefail
 
+FSTYPE="$(stat --file-system --format "%T" /)"
```

Comment:
Hmm, the test always creates a new image that uses btrfs? Does this mean that mkosi doesn't make use of that?

Reply:
@YHNdnzj Indeed, we don't reuse any of the existing bash framework. The goal is to replace it.

Reply:
> we don't reuse any of the existing bash framework. The goal is to replace it.

@DaanDeMeyer could you expand on that? The fuzzers I use to send packets to catch stuff like https://github.com/systemd/systemd/issues/31485, https://github.com/systemd/systemd/issues/30928 and so on kind of rely on the bash framework so set things up with some patches on top to make valigrind work so it would really be a bummer if it stops working because the bash framework is gone. I mean I can probably switch to the new thing but it would be great to figure out what the plan is.

Reply:
@evverx We're slowly getting all the integration tests running in github actions with meson + mkosi. The details are in test/meson.build and test/integration-test-wrapper.py and some docs in README.testsuite. There's still lots of functionality that is not implemented so we're not going to remove the bash stuff anytime soon.

The idea is to run the tests in images that mimick the actual distributions more compared to the adhoc image assembled by the bash framework. By using mkosi, it also becomes possible to run the tests for any distribution regardless of what distribution the host machine is running.

I'm happy to work with you to get the fuzzing stuff working on top of mkosi.

Reply:
Got it. Thanks! I'll take a look.

> run the tests in images that mimick the actual distributions more compared to the adhoc image assembled by the bash framework

Last time I checked it was kind of problematic to assemble mkosi images to run other instrumented components like dbus-broker (https://github.com/bus1/dbus-broker/issues/291) alongside systemd there. I agree that it's not what the bash framework is for but it is what it is :-)